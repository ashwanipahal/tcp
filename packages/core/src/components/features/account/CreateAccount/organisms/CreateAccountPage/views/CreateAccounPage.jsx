import React from 'react';
import PropTypes from 'prop-types';
import { getLabelValue } from '@tcp/core/src/utils/utils';
import Notification from '../../../../../../common/molecules/Notification';
import withStyles from '../../../../../../common/hoc/withStyles';
import styles from '../styles/CreateAccounPage.style';
import CreateAccountForm from '../../../molecules/CreateAccountForm';
import PasswordRequirement from '../../../../ResetPassword/molecules/PasswordRequirement';
import CreateAccountTopSection from '../../../molecules/CreateAccountTopSection';
import BodyCopy from '../../../../../../common/atoms/BodyCopy';
import SpinnerOverlay from '../../../../../../common/atoms/SpinnerOverlay';

class CreateAccounPage extends React.Component {
  static propTypes = {
    createAccountAction: PropTypes.func,
    className: PropTypes.string,
    labels: PropTypes.shape({}),
    isIAgreeChecked: PropTypes.bool,
    hideShowPwd: PropTypes.bool,
    confirmHideShowPwd: PropTypes.bool,
    error: PropTypes.string,
    onAlreadyHaveAnAccountClick: PropTypes.func,
    showForgotPasswordForm: PropTypes.func,
    isUserLoggedIn: PropTypes.bool.isRequired,
    formErrorMessage: PropTypes.shape({}).isRequired,
    userplccCardNumber: PropTypes.string.isRequired,
    userplccCardId: PropTypes.string.isRequired,
  };

  static defaultProps = {
    createAccountAction: () => {},
    className: '',
    labels: {
      lbl_createAccount_hide: 'hide',
    },
    isIAgreeChecked: false,
    hideShowPwd: false,
    confirmHideShowPwd: false,
    error: {},
    onAlreadyHaveAnAccountClick: () => {},
    showForgotPasswordForm: () => {},
  };

  constructor(props) {
    super(props);
    this.handleSubmitForm = this.handleSubmitForm.bind(this);
  }

  componentDidUpdate(prevProps) {
    const { isUserLoggedIn, error } = this.props;
    const container = document.getElementById('dialogContent');

    // Scrolling up to see the success end error message
    if (((!prevProps.isUserLoggedIn && isUserLoggedIn) || error) && container) {
      container.scrollTo(0, 0);
    }
  }

  handleSubmitForm(payload) {
    const { createAccountAction } = this.props;
    createAccountAction(payload);
  }

  render() {
    const {
      className,
      labels,
      isIAgreeChecked,
      hideShowPwd,
      confirmHideShowPwd,
      error,
      onAlreadyHaveAnAccountClick,
      showForgotPasswordForm,
      isUserLoggedIn,
      formErrorMessage,
      userplccCardNumber,
      userplccCardId,
    } = this.props;
    return (
      <div className={className}>
        <div className="parent-wrapper">
          <CreateAccountTopSection
            labels={labels}
            showForgotPasswordForm={showForgotPasswordForm}
          />
          {error && !isUserLoggedIn && (
            <div className="elem-pl-LRG elem-pr-LRG elem-pt-LRG">
              <BodyCopy
                fontFamily={['secondary']}
                fontWeight="semibold"
                fontSize="fs12"
                color="error"
              >
                {error}
              </BodyCopy>
            </div>
          )}

          {isUserLoggedIn && (
            <div className="elem-pl-LRG elem-pr-LRG elem-pt-LRG">
              <Notification
                status="success"
                colSize={{ large: 12, medium: 8, small: 6 }}
                message={getLabelValue(labels, 'lbl_createAccount_succcessMsg', 'registration')}
              />
            </div>
          )}

          <CreateAccountForm
            className={className}
            labels={labels}
            onSubmit={this.handleSubmitForm}
            isIAgreeChecked={isIAgreeChecked}
            hideShowPwd={hideShowPwd}
            initialValues={{ rememberMe: true }}
            confirmHideShowPwd={confirmHideShowPwd}
            onAlreadyHaveAnAccountClick={onAlreadyHaveAnAccountClick}
            tooltipContent={<PasswordRequirement labels={labels.password} />}
            formErrorMessage={formErrorMessage}
            userplccCardNumber={userplccCardNumber}
            userplccCardId={userplccCardId}
          />
        </div>
      </div>
    );
  }
}

export default withStyles(CreateAccounPage, styles);
export { CreateAccounPage as CreateAccounPageVanilla };
