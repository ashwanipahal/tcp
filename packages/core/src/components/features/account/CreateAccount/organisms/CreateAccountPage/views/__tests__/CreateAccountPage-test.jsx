import React from 'react';
import { shallow } from 'enzyme';
import { CreateAccounPageVanilla } from '../CreateAccounPage';

describe('Create Account Page', () => {
  it('should render correctly', () => {
    const mockedcreateAccountAction = jest.fn();
    const tree = shallow(
      <CreateAccounPageVanilla
        createAccountAction={mockedcreateAccountAction}
        labels={{ CREATE_ACC_LBL_HIDE: 'hide' }}
        className=""
      />
    );
    tree.instance().handleSubmitForm();
    expect(mockedcreateAccountAction).toBeCalled();
    expect(tree).toMatchSnapshot();
  });
});
