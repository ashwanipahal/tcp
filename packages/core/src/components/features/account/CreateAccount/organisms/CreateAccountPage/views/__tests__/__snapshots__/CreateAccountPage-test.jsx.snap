// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Account Page should render correctly 1`] = `
<div
  className=""
>
  <div
    className="parent-wrapper"
  >
    <Styled(CreateAccountTopSection)
      labels={
        Object {
          "lbl_createAccount_hide": "hide",
          "registration": Object {},
        }
      }
      showForgotPasswordForm={[Function]}
    />
    <div
      className="elem-pl-LRG elem-pr-LRG elem-pt-LRG"
    >
      <Styled(BodyCopy)
        className={null}
        color="error"
        component="p"
        fontFamily={
          Array [
            "secondary",
          ]
        }
        fontSize="fs12"
        fontWeight="semibold"
        letterSpacing="normal"
        lineHeight="normal"
        textAlign="left"
      >
        <Component />
      </Styled(BodyCopy)>
    </div>
    <Styled(ReduxForm)
      className=""
      confirmHideShowPwd={false}
      destroyOnUnmount={true}
      enableReinitialize={true}
      forceUnregisterOnUnmount={false}
      form="CreateAccountForm"
      getFormState={[Function]}
      hideShowPwd={false}
      initialValues={
        Object {
          "rememberMe": true,
        }
      }
      isIAgreeChecked={false}
      keepDirtyOnReinitialize={false}
      labels={
        Object {
          "lbl_createAccount_hide": "hide",
          "registration": Object {},
        }
      }
      onAlreadyHaveAnAccountClick={[Function]}
      onSubmit={[Function]}
      persistentSubmitErrors={false}
      pure={true}
      shouldAsyncValidate={[Function]}
      shouldError={[Function]}
      shouldValidate={[Function]}
      shouldWarn={[Function]}
      submitAsSideEffect={false}
      tooltipContent={<PasswordRequirement />}
      touchOnBlur={true}
      touchOnChange={false}
      updateUnregisteredFields={false}
      validate={[Function]}
    />
  </div>
</div>
`;
